src/services/spheres.js

```diff
@@ -178,8 +178,21 @@ class Spheres {
       detailedLocation,
     );
 
+    let chartForIsland;
+    if (!_.isNil(itemAtLocation)
+        && LogicHelper.isRandomizedCharts()
+        && itemAtLocation.match(/(Treasure|Triforce) Chart (\d)+/)) {
```

Comment:
This line should be added as a method to `LogicHelper`, since we're now using it in two places.